source/src/protocols/protein_grid/ProteinGrid.cc

```diff
@@ -0,0 +1,1493 @@
+	if ( sub_region_dimensions.size() != 3 ) {
+		ms_tr.Warning << "Sub region dimensions vector that was provided does not have 3 expected values. Contents of vector: [";
+		for ( auto val : sub_region_dimensions ) {
+			ms_tr.Warning << val << ",";
+		}
+		ms_tr.Warning << "]" << std::endl;
+	}
+
+	//set values of true/absolute sub areas (center and dimensions before shifting)
+	true_sub_area_center_ = sub_area_center;
+	true_sub_region_dimensions_ = sub_region_dimensions;
+
+	using_sub_area_ = true;
+
+	//wrap matrix around pose
+	wrap_matrix_around_pose();
+}
+
+//copy constructor
+ProteinGrid::ProteinGrid( ProteinGrid const & other )
+{
+	initialize();
+	other.clone( *this );
+}
+
+//@brief = operator overload
+ProteinGrid & ProteinGrid::operator=( ProteinGrid const & other ) {
+	other.clone( *this );
+	return *this;
+}
+
+// destructor
+ProteinGrid::~ProteinGrid() = default;
+
+// @brief function to clone the current ProteinGrid into inputted ProteinGrid
+void ProteinGrid::clone(ProteinGrid & copy) const {
+	// copy over all member variables
+	copy.protein_matrix_ = this->protein_matrix_;
+	copy.working_pose_ = this->working_pose_;
+	copy.xyz_shift_ = this->xyz_shift_;
+	copy.xyz_bound_ = this->xyz_bound_;
+	copy.matrix_volume_ = this->matrix_volume_;
+	copy.sub_matrix_volume_ = this->sub_matrix_volume_;
+	copy.resolution_ = this->resolution_;
+	copy.using_sub_area_ = this->using_sub_area_;
+	copy.using_lj_radii_ = this->using_lj_radii_;
+	copy.true_sub_area_center_ = this->true_sub_area_center_;
+	copy.adjusted_sub_area_center_ = this->adjusted_sub_area_center_;
+	copy.true_sub_region_dimensions_ = this->true_sub_region_dimensions_;
+	copy.adjusted_sub_region_dimensions_ = this->adjusted_sub_region_dimensions_;
+	copy.sub_region_max_ = this->sub_region_max_;
+	copy.sub_region_min_ = this->sub_region_min_;
+	copy.matrix_fullness_ = this->matrix_fullness_;
+	copy.fullness_ratio_ = this->fullness_ratio_;
+	copy.sub_matrix_fullness_ = this->sub_matrix_fullness_;
+	copy.sub_fullness_ratio_ = this->sub_fullness_ratio_;
+	copy.print_whole_matrix_ = this->print_whole_matrix_;
+	copy.print_empty_space_ = this->print_empty_space_;
+}
+
+// @brief function to initialize member variables at constructor calls
+//this sets initial and default values for member variables that we can build the matrix upon
+void ProteinGrid::initialize()
+{
+
+	matrix_volume_ = 0;
+	sub_matrix_volume_ = 0;
+	resolution_ = 1;
+	using_sub_area_ = false;
+	using_lj_radii_ = false;
+	matrix_fullness_ = 0;
+	fullness_ratio_ = 0;
+	sub_matrix_fullness_ = 0;
+	sub_fullness_ratio_ = 0;
+
+	//seed variables that are relevant for how to output the matrix to a pdb for visualization
+	//not planning to have devoted constructors to define these variables
+	print_whole_matrix_ = basic::options::option[ basic::options::OptionKeys::protein_grid::output_whole_matrix ]();
+	print_empty_space_ = basic::options::option[ basic::options::OptionKeys::protein_grid::output_empty_space ]();
```

Comment:
Consulting the option system isn't possible with in-class initialization, so setting these with a function make sense, though convention is to be explicit in the function name (i.e. `initialize_from_options()`)